src/registrar/views/domain.py

```diff
@@ -96,6 +96,55 @@ def form_valid(self, formset):
         return super().form_valid(formset)
 
 
+class DomainSecurityEmailView(DomainPermission, FormMixin, DetailView):
+
+    """Domain security email editing view."""
+
+    model = Domain
+    template_name = "domain_security_email.html"
+    context_object_name = "domain"
+    form_class = DomainSecurityEmailForm
+
+    def get_initial(self):
+        """The initial value for the form."""
+        domain = self.get_object()
+        initial = super().get_initial()
+        initial["security_email"] = domain.security_email()
+        return initial
+
+    def get_success_url(self):
+        """Redirect to the overview page for the domain."""
+        return reverse("domain-security-email", kwargs={"pk": self.object.pk})
+
+    def post(self, request, *args, **kwargs):
+        """Formset submission posts to this view."""
+        self.object = self.get_object()
+        form = self.get_form()
+        if form.is_valid():
+            # there is a valid email address in the form
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+    def form_valid(self, form):
+        """The form is valid, call setter in model."""
+
+        # Set the security email from the from
+        try:
+            new_email = form.cleaned_data["security_email"]
+        except KeyError:
+            # no server information in this field, skip it
+            pass
+        domain = self.get_object()
+        domain.set_security_email(new_email)
```

Comment:
You used new_email here but if it fails on the try new_email will have never been set.   In the except you can set new_email to none if that is the intention and requirement if a security email doesn't exist. Or just use a else after the except and only set the domain email if there is in fact an email available to use.

Reply:
You're right, that pass does not make sense. The exception should not happen so maybe that whole try/except block is unneeded and can be replaced by new_email = form.cleaned_data["security_email"]?

Reply:
I would say, if there is a chance of a crashing exception on the server side, best practice is to catch it. So I wouldn't remove the try/except; for consistency you could follow this pattern where you set new_email to have a default value on error. 
![image](https://github.com/cisagov/getgov/assets/109625347/52e1fa6e-4e51-45f2-9023-3a8512400a32)
or 
just move the domain.set_security_email portion into an else. The else means "If the exception is not hit", so no email gets added if there is no email (which sounds like that shouldn't happen really anyways).
```
       else:
              domain = self.get_object()
              domain.set_security_email(new_email)
```

That being said I saw another part of the code not doing the try/except block too, so I am fine with it being dealers choice and we can circle back to it if we ever need to.

Reply:
Let's see if 3rd time's the charm. I removed the 'KeyError' on the except as it seemed like it's specific to the Neil's use case and initialized new_email to an empty string.